run-tests.sh

```diff
@@ -14,14 +50,42 @@ do
   full_second_level_dir=$(echo "$dir" | cut -d/ -f1-2)
 
   # Only run if change happened at second level, since first level-changes don't require the tests to be re-run.
+  # If change happened at first level, $second_level_dir will be empty.
   if [ ! -z "$second_level_dir" -a "$second_level_dir" != " " ]; then
     if [[ ! " ${tested_dirs[*]} " =~ " ${full_second_level_dir} " ]]; then
+      echo "Executing tests for $full_second_level_dir"
+
       tested_dirs+=($full_second_level_dir)
       if [ -e $full_second_level_dir/requirements.txt ]; then
-        python -m pip install -r $full_second_level_dir/requirements.txt --extra-index-url https://download.pytorch.org/whl/cpu
+        $1 -m pip -q install -r $full_second_level_dir/requirements.txt --extra-index-url https://download.pytorch.org/whl/cpu --no-warn-script-location
+      fi
+
+      # Ensure proper spaCy version is installed.
+      spacy_version=$(
+        cat ${full_second_level_dir}/project.yml |
+        $1 -c "import yaml; import sys; print(yaml.safe_load(sys.stdin.read()).get('spacy_version', ''))"
```

Comment:
I don't really understand why you don't just use `srsly` here? You can add it to the list of things that aren't uninstalled like `torch`?

Reply:
Switched to `srsly`.